kw6/reader.py

```diff
@@ -40,37 +40,42 @@ class Config:
         arbitrary_types_allowed = True
 
     @staticmethod
-    @validate_arguments
-    def from_path(path: Path, header_path: Path = None):
-        stream = path.open("rb")
-
-        version = stream.read(settings.N_BYTES_VERSION).decode().strip()
+    def from_file_like(file, header_file=None):
+        version = file.read(settings.N_BYTES_VERSION).decode().strip()
         if version != "KW6FileClassVer1.0":
-            raise ValueError(f"Unexpected file version {version} {path}")
+            raise ValueError(f"Unexpected file version {version}")
 
-        initial_position_header = PositionHeader.from_stream_(stream)
+        initial_position_header = PositionHeader.from_stream_(file)
 
         cached_byte_positions = (
-            header.positions(header_path)
-            if header_path is not None
-            else dict()
+            dict() if header_file is None else header.positions(header_file.read())
         )
         cached_byte_positions[initial_position_header.frame_index] = settings.N_BYTES_VERSION
 
-        stream.seek(0, io.SEEK_END)
-        n_bytes = stream.tell()
+        file.seek(0, io.SEEK_END)
+        n_bytes = file.tell()
 
         return Reader(
-            path=path,
-            stream=stream,
+            path=None,
+            stream=file,
             cached_byte_positions=cached_byte_positions,
             initial_frame_index=initial_position_header.frame_index,
             n_bytes=n_bytes,
         )
 
+    @staticmethod
+    @validate_arguments
+    def from_path(path: Path, header_path: Path = None):
+        reader = Reader.from_file_like(
+            path.open("rb"),
+            None if header_path is None else header_path.open("rb"),
+        )
+        reader.path = path
+        return reader
+
     def __iter__(self):
         """Iterate over all positions and cameras in the file"""
-        stream = self.path.open("rb")
+        stream = (self.stream if self.path is None else self.path.open("rb"))
```

Comment:
Cant we just do `self.stream` and remove the `self.path` attribute entirely? I have some memory of this not working for some reason so correct me if I'm wrong